File: solutions/php/darts/1/Darts.php
<?php

declare(strict_types=1);

class Darts
{
    public int $score = 0;
    private const INNER_RADIUS = 1;
    private const INNER_SCORE = 10;
    private const MIDDLE_RADIUS = 5;
    private const MIDDLE_SCORE = 5;
    private const OUTER_RADIUS = 10;
    private const OUTER_SCORE = 1;

    public function __construct(float $xAxis, float $yAxis)
    {
        $distance = sqrt($xAxis ** 2 + $yAxis ** 2);

        $this->score = match (true) {
            $distance <= self::INNER_RADIUS => self::INNER_SCORE,
            $distance <= self::MIDDLE_RADIUS => self::MIDDLE_SCORE,
            $distance <= self::OUTER_RADIUS => self::OUTER_SCORE,
            default => 0,
        };
    }

}

File: solutions/python/darts/1/darts.py
from math import sqrt

def score(x: int | float, y: int | float) -> int:
    match sqrt(x ** 2 + y ** 2):
        case d if 0 <= d <= 1:
            return 10
        case d if 1 < d <= 5:
            return 5
        case d if 5 < d <= 10:
            return 1
        case _:
            return 0
